sympy/ntheory/digits.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import print_function, division
+
+import math
+import re
+
+from collections import (
+    Counter,
+    OrderedDict,
+)
+
+
+def count_digits(n, base=10):
```

Comment:
I'm not sure we need this "battery" to be added when `multiset(digits(n, base)[1:])` will do. If it must be sorted then `OrderedDict([(k, v in sorted(multiset(digits(n, base)[1:]).items())])`. But it might be better to use a `defaultdict(int)` so a request for a digit that is not present will return 0:
```python
>>> d = defaultdict(int, multiset(digits(123, 10)[1:]).items())
>>> d[4]
0
```

Reply:
If you don't think it's required I can remove it (and the test), but the point surely is that a user may not know of such a concise way of obtaining the digit frequency count, and may want to do this via SymPy, as with `digits`, for which there are one-liners also.

Reply:
I've updated `count_digits` to use `defaultdict`, and also the tests.